modules/apigee_edge_teams/src/User/RemoveTeamRolesWithUserSynchronousUserRemovalHandler.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Copyright 2023 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_teams\User;
+
+use Drupal\apigee_edge\User\UserRemovalHandlerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Utility\Error;
+use Drupal\user\UserInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+
+/**
+ * Ensures team roles of the removed user also get deleted.
+ */
+final class RemoveTeamRolesWithUserSynchronousUserRemovalHandler implements UserRemovalHandlerInterface {
+
+  /**
+   * RemoveTeamRolesWithUserSynchronousUserRemovalHandler constructor.
+   */
+  public function __construct(private readonly UserRemovalHandlerInterface $decorated, private readonly EntityTypeManagerInterface $entityTypeManager, private readonly LoggerInterface $logger) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __invoke(UserInterface $account): void {
+    ($this->decorated)($account);
```

Comment:
Double checked and this was the original order as well, the hook in apigee_edge was called sooner than in this module - which makes total sense.